### ISSUE TITLE
Style guide

### ISSUE THREAD

**Issue by eliascodes** [foundersandcoders/master-reference#788]
As an outcome of #735, we've decided we need a style guide in addition to the contributing guide. The latter should address process, the former content and ...style. Not sure whether these need to be separate docs, but we'll go with that for now.

@astroash thanks for your [initial efforts](https://hackmd.io/IYVgpgzAHAnFwFoCMMQDYEBYoBMDGCARumAjgGaEBMOADCLTIXlEA===?view). Opening this so we don't pollute the other issue.

Feel free to open a PR with this so we can comment on the document directly.

FYI @sofer @des-des @shiryz @m4v15 

### General comments
IMO, there are two main elements of the curriculum: weeks and resources (workshops/MCs/slides/etc). The style guide should address both the creation and maintenance of both of these. 

#### Weeks
Some of the course learning outcomes are achieved through structural elements of the week (projects, research, etc.). The style guide should make recommendations about what elements we currently use, but I feel like in principle we should be able to adopt different (new) elements if we can justify them in terms of the course outcomes.

For creating a new week (like e.g. express week), we should offer guidance for how to go about designing it.

#### Resources
Where new resources are created, I think that we should encourage a more iterative/consultative process where the people creating the resource seek feedback regularly, particularly at the outset, to avoid wasted effort.

We should also give guidance on how to structure workshops (or MCs) with reference to both LO's and also examples of workshops (or MCs) that have previously been successful. For example, encourage resources that are focussed, interactive, require teamwork, etc. As mentioned by @ZooeyMiller [here](https://github.com/foundersandcoders/master-reference/issues/691#issuecomment-345607276), this may require/benefit from more mentor prep or mentor/delivery notes.

### Your questions
To answer your questions from the hackmd @astroash
> > link to the CONTRIBUTING.md in master-reference
>
> (currently removed from _Creation of New Resources_)  
> Does this really happen? Does it need to be added to every resource? This document also currently needs updating! Is this document replacing that one? I'll be honest I didn't read the whole thing...

This line was added very recently. It's a shortcut to avoid having to write a whole new CONTRIBUTING.md for every resource. See #668. The CONTRIBUTING.md definitely needs updating in parallel. See the first paragraph for the relation b/w the style guide and contributing doc.

> Week 4 is cute and has a [summary doc](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/week-4/summary.md), is this something we want to keep or implement in other weeks?

I suspect this was because node week represents a bit of a step change for students and mentors felt some kind of context/cheat-sheet would be beneficial. I'm agnostic.

> Can I cover this in emojis? :cactus: :japanese_ogre: :warning: :vhs: 

👍 

> layout - I ended up putting things in file structure order, but was also tempted to put lo stuff together/ naming stuff together, but this is probably better?

Could you explain this? Don't understand.

#### "IMO stuff"
Looks good.

**Comment by eliascodes** [comment]
We should cannibalise the existing guide on [planning-weeks](https://github.com/foundersandcoders/master-reference/blob/master/curriculum-planning/planning-weeks.md)

**Comment by astroash** [comment]
> The style guide should make recommendations about what elements we currently use, but I feel like in principle we should be able to adopt different (new) elements if we can justify them in terms of the course outcomes.

@eliascodes do you imagine this looking like the guidance in `planning-weeks.md`? This doc is way too wordy for my liking, I'm not sure what a half way would be like. Do you think it would be enough for _Creation of New Resources_ section having a subsection about 'New Weeks' which 3/4 key points from the `planning-weeks.md`?

> Where new resources are created, I think that we should encourage a more iterative/consultative process where the people creating the resource seek feedback regularly, particularly at the outset, to avoid wasted effort.

I think this is useful but should be covered in the curriculum update process outline rather than in this doc.

> We should also give guidance on how to structure workshops (or MCs) with reference to both LO's and also examples of workshops (or MCs) that have previously been successful. 

Is it worth opening another issue to ask for community feedback on which workshops they feel are successful and building these points from there?

____
Also feel free to add comments on my PR/ push to my branch with updates to the doc.

**Comment by eliascodes** [comment]
> This doc is way too wordy for my liking
Agree, I think the headings, plus one explanatory sentence should be enough for now.

> I think this is useful but should be covered in the curriculum update process outline rather than in this doc.
Maybe both?

> Is it worth opening another issue to ask for community feedback on which workshops they feel are successful and building these points from there?
Probably, though we're a bit short on time now.

I think the main thing that's missing is guidance on tone/level of guidance in resources. We need to give an indication as to how much detail in a resource is appropriate, rather than leaving it totally up to the personal preferences of whoever's doing the update.

**Comment by astroash** [comment]
@eliascodes can this be closed?
